pkg/registry/flowcontrol/flowschema/strategy.go

```diff
@@ -94,7 +94,10 @@ var StatusStrategy = flowSchemaStatusStrategy{Strategy}
 func (flowSchemaStatusStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
 	newFlowSchema := obj.(*flowcontrol.FlowSchema)
 	oldFlowSchema := old.(*flowcontrol.FlowSchema)
+
+	managedFields := newFlowSchema.ManagedFields
 	newFlowSchema.ObjectMeta = oldFlowSchema.ObjectMeta
+	newFlowSchema.ManagedFields = managedFields
```

Comment:
this is delicate enough that this needs a comment explaining what's going on (same for other places)

Reply:
do we have any tests that ensure an API caller cannot mess with managedFields directly (i.e. that the managedFields in the new object at this point is computed by the field manager based on the old object's managedFields and `diff(old,new)`)?

Reply:
As discussed, we don't, this is a known limitation. We should document that, and also try to provide a fix asap.

Reply:
Belated code review - this pattern would be a lot more clear if spelled as `temp := newObject.ObjectMeta.ManagedFields`.